### DISCUSSION TITLE
CSLA 9 release candidate 1 online

### DISCUSSION THREAD

**Discussion by rockfordlhotka** [MarimerLLC/csla#4383] [Announcements]
Hello everyone!

I am happy to announce that CSLA 9 release candidate 1 is now available in NuGet.

We are at the point with version 9 where the only changes expected are to fix bugs found by you, or as we update the samples in the `/Samples` directory. Other than bug fixes, CSLA 9 is complete and ready to go.

Please take the time to check it out and create issues here in GitHub if you find any bugs.

Here are the release notes: https://github.com/MarimerLLC/csla/blob/main/releasenotes.md

Here is a comprehensive list of changes: https://github.com/MarimerLLC/csla/compare/v8.2.7...HEAD

Check out the [Upgrading to CSLA 9](https://github.com/MarimerLLC/csla/blob/main/docs/Upgrading%20to%20CSLA%209.md) document.

Don't forget to check out the new `Csla.Generator` packages. One provides much higher performance serialization, and the other enables _radical_ code reduction in many business domain class scenarios.

* [AutoImplementProperties](https://github.com/MarimerLLC/csla/tree/main/Source/Csla.Generators/cs/AutoImplementProperties)
* [AutoSerialization](https://github.com/MarimerLLC/csla/tree/main/Source/Csla.Generators/cs/AutoSerialization)

**Comment by ossendorf-at-hoelscher** [comment]
Hi. I've just updated our CSLA from 8 to the 9-RC1.
I encountered the following issues/compilation errors:
* The new `Csla.NonSerializedAttribute` is now ambigious with the `System.NonSerializedAttribute`. Which poses the question: Which do I need to mark the field as not participating in the serialization process? (It was "introduced" in #4197 )
* The type `Csla.Serialization.SerializationFormatterFactory` got removed which is not documented in the CSLA 9 upgrade guide. It was removed with #4024 . I think the guide should say use `ApplicationContext.GetRequiredService<ISerializationFormatter>()` instead.
* `InputProperties` of rules are now read-only. Do we need to include that in the migration guide? So instead of assigning them just use `Add`/`AddRange` 🤔
* We should mention that the API of `IParent.RemoveChild` changed from `void` to `Task`. I catched that because we have promoted "not awaited calls" to errors which is not the default.

I'll update this post when I encounter other problems.

**Reply by rockfordlhotka** [reply]
Thank you, I don't think we considered that issue.

I am also not sure we need to have that attribute as a pollyfill since I very much doubt Microsoft will remove _their_ attribute due to the widespread issues that would cause people.

**Reply by rockfordlhotka** [reply]
Oh, no, it isn't a pollyfill. The issue is the applicable scope of the original attribute and our need to apply it to properties instead of fields.

We may just need to use a different name for our attribute to avoid the conflict.

**Reply by ossendorf-at-hoelscher** [reply]
Two more compile errors added

**Comment by ossendorf-at-hoelscher** [comment]
I also get this during compilation but I don't know if it was there before:
![image](https://github.com/user-attachments/assets/c4f9754d-e158-46e5-8623-64ea62a9068b)

**Reply by ossendorf-at-hoelscher** [reply]
I also get a `value can not be null` `ArgumentNullException` for principal when starting our application. I hope I can investigate this soon, but holidays are just around the corner 😅

**Reply by ossendorf-at-hoelscher** [reply]
Just created #4410 with the `ArgumentNullException` mentioned. Sadly that's a blocker for me to test anything further.

**Reply by ossendorf-at-hoelscher** [reply]
The analyzer exceptions are still a thing during compilation with rc3.

**Comment by ossendorf-at-hoelscher** [comment]
With the configuration changes made in v9 there is now no `IContextManager` registered when using this setup:
```csharp
.AddCsla(
    o => o.DataPortal(
        dpo => dpo.AddClientSideDataPortal(
            cso => cso.UseLocalProxy(lp => lp.UseLocalScope = false)
        )
    )
)
```
This is our current setup for our testing.
But that now fails because I can't retrieve an `IContextManager` to set my expected users.

Is there any recommendation on how to deal with that? Should I just use `AddConsoleApp` to get a context manager registered?

**Reply by rockfordlhotka** [reply]
I think that might be best.

Or we could register a default.
